lib/orbf/rules_engine/data/contract.rb

```diff
@@ -27,6 +27,10 @@ def distance(period)
         [start_period.to_i - period_start_month.to_i, end_period.to_i - period_start_month.to_i].min
       end
 
+      def org_unit
+        Orbf::RulesEngine::OrgUnit.new(ext_id: @org_unit["id"], name: @org_unit["name"], path: @org_unit["path"], group_ext_ids: [])
```

Comment:
Maybe move this into the initializer, so that we always use the `OrgUnit` model instead of the hash based `@org_unit`?

Reply:
While I thought you were right in the cleanness. Actually for garbage collection its worst we instantiate 6672 contracts but only call org_unit on a few of them (matching the period and main_contract_orgunit or "groups" fields).